### PR TITLE
OCPBUGS-10230: Fixes card sizes not even issue when commit info is not available on Environments page

### PR DESCRIPTION
**Pull request by ciiay**
Fixes [[gitops-2476: [UI] Card size is not the same when commit information is not available](https://issues.redhat.com/browse/GITOPS-2476).

signed by: [yicai@redhat.com](mailto:yicai@redhat.com)

Screenshot after fix:
Screenshot 2023-03-13 at 9 32 53 AM

### CODE REVIEW

File: frontend/packages/gitops-plugin/src/components/details/GitOpsDetails.tsx [hunk 1]
```diff
@@ -116,18 +116,26 @@ const GitOpsDetails: React.FC<GitOpsDetailsProps> = ({ envs, appName, manifestUR
...
                       {env.revision ? (
                         <>
-                          {env.revision.message && (
+                          {env.revision.message ? (
                             <StackItem className="gop-gitops-details__message">
                               {t('gitops-plugin~{{message}}', { message: env.revision.message })}
                             </StackItem>
+                          ) : (
+                            <StackItem className="gop-gitops-details__warning-message">
+                              <span>{t('gitops-plugin~Commit message not available')}</span>
```
Comment (debsmita1) [inline frontend/packages/gitops-plugin/src/components/details/GitOpsDetails.tsx:125]:
```suggestion
                              {t('gitops-plugin~Commit message not available')}
```

**Review by debsmita1 — CHANGES_REQUESTED** [review]

File: frontend/packages/gitops-plugin/locales/ja/gitops-plugin.json [hunk 2]
```diff
@@ -45,5 +45,8 @@
   "No GitOps manifest URLs found": "GitOps マニフェスト URL が見つかりません",
   "Error cannot retrieve applications": "エラー、アプリケーションを取得できません",
   "No Application groups found": "アプリケーショングループが見つかりません",
-  "Environment details were not found. Try reloading the page or contacting an administrator.": "環境の詳細が見つかりませんでした。ページのリロードを試行するか、管理者にお問い合わせください。"
+  "Environment details were not found. Try reloading the page or contacting an administrator.": "Environment details were not found. Try reloading the page or contacting an administrator.",
+  "Commit message not available": "コミットメッセージ情報はありません",
+  "Commit author not available": "コミット作成者情報はありません",
+  "Last deployed time not available": "最終デプロイ時刻情報はありません"
 }
```
Comment (lokanandaprabhu) [inline frontend/packages/gitops-plugin/locales/ja/gitops-plugin.json:52]: No need to add the translation other than English at this time.

File: frontend/packages/gitops-plugin/locales/ko/gitops-plugin.json [hunk 3]
```diff
@@ -45,5 +45,8 @@
   "No GitOps manifest URLs found": "GitOps 매니페스트 URL을 찾을 수 없습니다",
   "Error cannot retrieve applications": "오류로 인해 애플리케이션을 검색할 수 없습니다",
   "No Application groups found": "애플리케이션 그룹을 찾을 수 없습니다",
-  "Environment details were not found. Try reloading the page or contacting an administrator.": "환경 세부 정보를 찾을 수 없습니다. 페이지를 새로 고침하거나 관리자에게 문의하십시오."
+  "Environment details were not found. Try reloading the page or contacting an administrator.": "Environment details were not found. Try reloading the page or contacting an administrator.",
+  "Commit message not available": "커밋 메시지를 사용할 수 없음",
+  "Commit author not available": "커밋 작성자를 사용할 수 없음",
+  "Last deployed time not available": "마지막 배포 시간을 알 수 없음"
 }
```
Comment (lokanandaprabhu) [inline frontend/packages/gitops-plugin/locales/ko/gitops-plugin.json:52]: No need to add the translation other than English at this time.

File: frontend/packages/gitops-plugin/locales/zh/gitops-plugin.json [hunk 4]
```diff
@@ -45,5 +45,8 @@
   "No GitOps manifest URLs found": "未找到 GitOps 清单 URL",
   "Error cannot retrieve applications": "错误，无法检索应用程序",
   "No Application groups found": "未找到应用程序组",
-  "Environment details were not found. Try reloading the page or contacting an administrator.": "无法找到环境详情。请重新载入页面或联系管理员。"
+  "Environment details were not found. Try reloading the page or contacting an administrator.": "Environment details were not found. Try reloading the page or contacting an administrator.",
+  "Commit message not available": "提交的信息不可用",
+  "Commit author not available": "提交的作者不可用",
+  "Last deployed time not available": "最后部署的时间不可用"
```
Comment (lokanandaprabhu) [inline frontend/packages/gitops-plugin/locales/zh/gitops-plugin.json:51]: No need to add the translation other than English at this time.

File: frontend/packages/gitops-plugin/src/components/details/GitOpsDetails.tsx [hunk 5]
```diff
@@ -136,17 +144,30 @@ const GitOpsDetails: React.FC<GitOpsDetailsProps> = ({ envs, appName, manifestUR
...
-                      {env.lastDeployed && (
+                      {env.lastDeployed ? (
                         <StackItem className="co-truncate co-nowrap gop-gitops-details__env-section__time">
                           {t('gitops-plugin~Last deployed')}&nbsp;
                           <Timestamp timestamp={env.lastDeployed} />
                         </StackItem>
+                      ) : (
+                        <StackItem className="co-truncate co-nowrap gop-gitops-details__env-section__time-unavailable">
+                          <span>{t('gitops-plugin~Last deployed time not available')}</span>
+                        </StackItem>
```
Comment (lokanandaprabhu) [inline frontend/packages/gitops-plugin/src/components/details/GitOpsDetails.tsx:170]: {t('gitops-plugin~Last deployed time not available')}

**Comment by lokanandaprabhu** [comment]
I cloned your ticket as https://issues.redhat.com/browse/OCPBUGS-10230 to follow our bug tracking and merge process.


/retitle OCPBUGS-10230: Fixes card sizes not even issue when commit info is not available on Environments page

**Comment by lokanandaprabhu** [comment]
Tested locally. 

Screenshot 2023-03-13 at 7 08 41 PM

/lgtm

**Review by jerolimov — APPROVED** [review]
/approve

**Pull request merged**